src/ui/components/CreateIdentifier/components/IdentifierStage1.tsx

```diff
@@ -44,6 +46,13 @@ const IdentifierStage1 = ({
   const [scannedConections, setScannedConnections] = useState<
     ConnectionShortDetails[]
   >([]);
+  const incomingRequest = useMemo(() => {
+    return !queueIncomingRequest.isProcessing
+      ? { id: "" }
+      : queueIncomingRequest.queues.length > 0
+        ? queueIncomingRequest.queues[0]
+        : { id: "" };
+  }, [queueIncomingRequest]);
```

Comment:
I can see that you're trying to make sure the "Share identifier" screen is closed on the receiver side if you get the pop-up to accept.

Personally I would not bother with this complexity because:
1. It's a complexity that might regress easily
2. We know we are changing it so that external notifications won't be in this queue and instead in the database
3. It's possible that in a 2-of-3 multi-sig, the initiator might initiate too quickly before the 3rd has scanned this QR code (though this is still an issue if they close it, so we need to tackle that later too)

If we strongly feel like we need this feature we can keep it but I don't think it adds that much value and once we change the notifications it will be a whole lot more complex, or at least inefficient to check for this.

Reply:
rest of PR LGTM

Reply:
Yes, the following code is making sure that if we receive a request and the `groupId`  is the same, this screen will be closed so the user can accept the request coming from the initiator sitting one layer below which is for the same multi sig.

What would you suggest using instead of that to trigger the modal to close @iFergal ?

```
  useEffect(() => {
    incomingRequest.multisigIcpDetails?.ourIdentifier.groupMetadata?.groupId ===
      groupId && handleDone();
  }, [groupMetadata, incomingRequest]);
```
 
![16052024114811](https://github.com/cardano-foundation/cf-identity-wallet/assets/119612231/0c00fefd-0f46-48cf-8fa7-63930207af03)

Reply:
Why is the incoming request a layer below? I thought this would be above everything except for a login screen.

Reply:
I thought there would be no use in keeping that modal open up until after the entire request is completed.

Reply:
There is a use case, which is number 3 in my list.

Anyway, I'm discussing with Robert now - either we will keep it as is and change it once the notifications change, or we will just remove it now and not need to change it later.

Reply:
>Why is the incoming request a layer below? I thought this would be above everything except for a login screen.

Right now Ionic/Capacitor is dynamically assigning z-index for every element in the app and if it's not strictly necessary we don't mess with z-index manually (because in the past we did and it was working for a while then it stopped because we can't guess an accurate number to make it work with what Ionic/Capacitor will dynamically assign every time being different).

In this case we didn't have to because that modal was being closed while the request was coming in. That said, I will pass this comment to Duke which is currently working on refactoring that component in [this PR](https://github.com/cardano-foundation/cf-identity-wallet/pull/474) and make sure that will always show on top of everything (as expected).

Reply:
Cool. As agreed on Slack, we can merge this but I'll create a ticket to have it removed at some point (either soon to satisfy number 3, or later when notifications change).